### PR TITLE
[GC] add API to get obj defined type

### PR DESCRIPTION
**Pull request by xujuntwt95329**

### CODE REVIEW

File: core/iwasm/common/gc/gc_common.c [hunk 1]
```diff
@@ -50,6 +50,46 @@ wasm_get_defined_type(WASMModuleCommon *const module, uint32 index)
...
+{
+    if ((!wasm_obj_is_struct_obj(obj)) && (!wasm_obj_is_array_obj(obj))
+        && (!wasm_obj_is_func_obj(obj))) {
+        bh_assert(false);
+    }
+
+    return ((WASMRttTypeRef)(obj->header))->defined_type;
+}
+
+int32_t
```
Comment (wenyongh) [inline core/iwasm/common/gc/gc_common.c:64]: Had better `int32`

File: core/iwasm/common/gc/gc_common.c [hunk 2]
```diff
@@ -50,6 +50,46 @@ wasm_get_defined_type(WASMModuleCommon *const module, uint32 index)
...
+wasm_obj_get_defined_type_idx(WASMModuleCommon *const module,
+                              const WASMObjectRef obj)
+{
+    WASMType *type = wasm_obj_get_defined_type(obj);
+    int32_t i, type_idx = -1;
```
Comment (wenyongh) [inline core/iwasm/common/gc/gc_common.c:69]: int32

File: core/iwasm/common/gc/gc_common.c [hunk 3]
```diff
@@ -444,6 +484,55 @@ wasm_func_obj_new_with_type(WASMExecEnv *exec_env, WASMFuncType *type)
     return NULL;
 }
 
+bool
+wasm_runtime_call_func_ref(WASMExecEnv *exec_env,
+                           const WASMFuncObjectRef func_obj, uint32_t argc,
+                           uint32_t argv[])
```
Comment (wenyongh) [inline core/iwasm/common/gc/gc_common.c:490]: `uint32 argc, uint32 argv[]`

Comment (xujuntwt95329) [reply core/iwasm/common/gc/gc_common.c]: Done, thanks~

**Pull request merged**